src/seal_case.c

```diff
@@ -0,0 +1,173 @@
+    return inventory->seals[sealId];
+}
+
+BOOL SealIsOnCapsule(const CAPSULE *capsule, int sealId) {
+    int i;
+
+    for (i = 0; i < MAX_SEALS_ON_CAPSULE; i++) {
+        if (sealId + 1 == capsule->seals[i].kind) {
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+static int SealCase_CountSealOccurrenceInUse(const SEALCASE *sealCase, int sealId) {
+    int i, j, result;
+
+    for (result = 0, i = 0; i < MAX_CAPSULES; i++) {
+        for (j = 0; j < MAX_SEALS_ON_CAPSULE; j++) {
+            if (sealId + 1 == sealCase->capsules[i].seals[j].kind) {
+                result++;
+            }
+        }
+    }
+
+    return result;
+}
+
+void SealCaseInventory_SetSealQuantity(SEALBAG *inventory, int sealId, s16 quantity) {
+    inventory->seals[sealId] = quantity;
+}
+
+BOOL GiveOrTakeSeal(SEALCASE *sealCase, int sealId, s16 quantity) {
+    int num;
+    int total;
+
+    num = SealCase_CountSealOccurrenceInUse(sealCase, sealId - SEAL_MIN);
+    total = num + sealCase->inventory.seals[sealId - SEAL_MIN];
+
+    if (quantity < 0) {
+        if (sealCase->inventory.seals[sealId - SEAL_MIN] + quantity >= 0) {
+            sealCase->inventory.seals[sealId - SEAL_MIN] += quantity;
+        } else {
+            return FALSE;
+        }
+    } else {
+        if (total + quantity <= SEAL_INVENTORY_QMAX) {
+            sealCase->inventory.seals[sealId - SEAL_MIN] += quantity;
+        } else {
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+BOOL GiveOrTakeSeal2(SEALCASE *sealCase, int sealId, s16 quantity) {
+    int num;
+    int total;
+
+    num = SealCase_CountSealOccurrenceInUse(sealCase, sealId - SEAL_MIN);
+    total = num;
+
+    if (quantity < 0) {
+        if (sealCase->inventory.seals[sealId - SEAL_MIN] + quantity >= 0) {
+            sealCase->inventory.seals[sealId - SEAL_MIN] += quantity;
+        } else {
+            return FALSE;
+        }
+    } else {
+        if (total + quantity <= SEAL_INVENTORY_QMAX) {
+            sealCase->inventory.seals[sealId - SEAL_MIN] += quantity;
+            if (sealCase->inventory.seals[sealId - SEAL_MIN] >= SEAL_INVENTORY_QMAX) {
+                sealCase->inventory.seals[sealId - SEAL_MIN] = SEAL_INVENTORY_QMAX;
+            }
+        } else {
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+BOOL SealCase_CheckSealQuantity(const SEALCASE *sealCase, int sealId, s16 quantity) {
+    int num;
+    int total;
+
+    num = SealCase_CountSealOccurrenceInUse(sealCase, sealId - SEAL_MIN);
+    total = num + sealCase->inventory.seals[sealId - SEAL_MIN];
+
+    if (quantity < 0) {
+        if (sealCase->inventory.seals[sealId - SEAL_MIN] + quantity >= 0) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
+    } else {
+        if (total + quantity <= SEAL_INVENTORY_QMAX) {
+            return TRUE;
+        } else {
+            return FALSE;
+        }
```

Comment:
```suggestion
        return (total + quantity) <= SEAL_INVENTORY_QMAX;
```

Reply:
Rejecting change to maintain similarity with the above two functions.